### PR TITLE
refac: Rename 'datamigration' folder to 'datamigration_hive'

### PR DESCRIPTION
**Pull request by JesperGomes**
# Description
NOTE: THIS PR SHOULD GO IN TOGETHER WITH THIS: https://github.com/Energinet-DataHub/dh3-environments/pull/1073/

Rename existing folder from 'datamigration' to 'datamigration_hive' and adding a new (almost) empty 'datamigration' folder

image

## Pull-request quality

- [ ] The title adheres to [this guide](https://github.com/Mech0z/GitHubGuidelines)
- [ ] Tests are written and executed locally
- [ ] Subsystem tests have been tested (by manually deploying to `dev_002`)

### CODE REVIEW

File: .github/workflows/detect-changes.yml [hunk 1]
```diff
@@ -70,15 +70,15 @@ jobs:
               - '.devcontainer/docker-compose.yml'
               - '.docker/requirements.txt'
             delta_migrations:
-              - 'source/databricks/calculation_engine/package/datamigration/**'
+              - 'source/databricks/calculation_engine/package/datamigration_hive/**'
```
Comment (BjarkeMeier) [inline .github/workflows/detect-changes.yml:73]: Should it be this?
```suggestion
              - 'source/databricks/calculation_engine/package/datamigration/**'
              - 'source/databricks/calculation_engine/package/datamigration_hive/**'
```

File: .github/workflows/detect-changes.yml [hunk 2]
```diff
@@ -70,15 +70,15 @@ jobs:
...
               - 'source/databricks/calculation_engine/tests/datamigration/**'
               - 'source/databricks/calculation_engine/tests/helpers/spark_sql_migration_helper.py'
             dotnet:
               - '.github/workflows/cd.yml'
               - '.github/workflows/ci-dotnet.yml'
               - '.github/actions/dotnet-*/**'
               - 'source/dotnet/!(Packages/**)/**'
-              - 'source/databricks/calculation_engine/package/datamigration/migration_scripts/**'
+              - 'source/databricks/calculation_engine/package/datamigration_hive/migration_scripts/**'
```
Comment (BjarkeMeier) [inline .github/workflows/detect-changes.yml:81]: Does .NET still depend on delta migrations?

**Review by BjarkeMeier — APPROVED** [review]

File: .github/workflows/detect-changes.yml (hunk 1 above)
Comment (JesperGomes) [reply .github/workflows/detect-changes.yml:73]: yes, I will ensure this is done afterwards

File: .github/workflows/detect-changes.yml (hunk 2 above)
Comment (JesperGomes) [reply .github/workflows/detect-changes.yml:81]: good question - I will find out

Comment (JesperGomes) [reply .github/workflows/detect-changes.yml:81]: Mosaic will investigate this

**Review by kerbou — APPROVED** [review]

**Pull request merged**